conn2res/plotting.py

```diff
@@ -0,0 +1,248 @@
+
+
+def plot_time_series(x, feature_set='orig', idx_features=None, n_features=None, xlim=[0, 150], ylim=None,
+                     scaler=1, num=1, figsize=(12, 6), subplot=None, title=None, fname='time_course',
+                     legend_label='', savefig=False, block=True, **kwargs):
+    # transform data
+    x = transform_data(
+        x, feature_set, idx_features=idx_features, n_features=n_features, scaler=scaler, **kwargs)
+
+    # open figure and create subplot
+    plt.figure(num=num, figsize=figsize)
+    if subplot is None:
+        subplot = (1, 1, 1)
+    plt.subplot(*subplot)
+
+    # plot data
+    if x.size > 0:
+        plt.plot(x)
+
+        # add x and y limits
+        plt.xlim(xlim)
+        if ylim is not None:
+            plt.xlim(ylim)
+
+        # plot legend
+        try:
+            legend = [f'{legend_label} {n+1}' for n in range(x.shape[1])]
+        except:
+            legend = [f'{legend_label} 1']
+        try:  # quick fix to get previously plotted legends
+            lg = plt.gca().lines[-1].axes.get_legend()
+            legend = [text.get_text() for text in lg.texts] + legend
+        except:
+            pass
+        if len(legend) <= 5:
+            plt.legend(legend, loc='upper right', fontsize=12)
+        else:
+            plt.legend(legend, loc='upper right', fontsize=12, ncol=2)
+        # plt.legend(legend, loc='upper center', bbox_to_anchor=(
+        #     0.5, 1.05), fontsize=10, ncol=len(legend))
+
+    # set xtick/ythick fontsize
+    plt.xticks(fontsize=22)
+    plt.yticks(fontsize=22)
+
+    # add title
+    if title is not None:
+        plt.title(f'{title} time course', fontsize=22)
+
+    # set tight layout in case there are different subplots
+    plt.tight_layout()
+
+    if savefig:
+        plt.savefig(fname=os.path.join(FIG_DIR, f'{fname}.png'),
+                    transparent=True, bbox_inches='tight', dpi=300)
+    plt.show(block=block)
+
+
+def transform_data(data, feature_set, idx_features=None, n_features=None, scaler=1, model=None, **kwargs):
+
+    if feature_set == 'pc':
+        # transform data into principal components
+        data = PCA(idx_pcs=idx_features,
+                   n_pcs=n_features).fit_transform(data, **kwargs)
+
+    elif feature_set == 'idx':
+        # select given features
+        data = data[:, idx_features]
+
+    elif feature_set == 'rnd':
+        # update default number of features
+        if n_features is None:
+            n_features = 1
+
+        # choose features randomly
+        data = data[:, np.random.choice(
+            np.arange(data.shape[1]), size=n_features)]
+
+    elif feature_set == 'decfun':
+        # calculate decision function using model fitted on time series
+        data = model.decision_function(data)
+
+    elif feature_set == 'coeff':
+        # get coefficient from model
+        if model.coef_.ndim > 1:
+            idx_class = kwargs.get('idx_class', 0)
+            coef = model.coef_[idx_class, :]
```

Comment:
In a multi-class/multi-output regression model.coef_ is 2D and I thought it is useful to plot the coefficients for all classes/outputs separately and now I set the default to plot the first class/output. But perhaps a better way would be some sort of averaging over classes/outputs.